src/workerd/jsg/modules-new.h

```diff
@@ -171,9 +171,25 @@ class Module {
     // This is generally only suitable for worker-bundle entry point modules,
     // but could in theory be applied to any module.
     MAIN = 1 << 0,
+    // A Module with the ESM flag set is interpreted as an ECMAScript module.
     ESM = 1 << 1,
+    // A Module with the EVAL flag set is interpreted as a module that requires
```

Comment:
The description here is a bit confusing – with the current implementation `EVAL` being set only means that we use `maybeEvalCallback` if present, there is no error if there is no callback available. Maybe describing it with something like "if Flag::EVAL is not set, then the EvalCallback is ignored" (i.e. the converse of what we state below) would be more precise.